flow-tests/test-themes/frontend/my-field.js

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2000-2020 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import '@vaadin/vaadin-lumo-styles/color.js';
+import '@vaadin/vaadin-lumo-styles/sizing.js';
+import '@vaadin/vaadin-lumo-styles/spacing.js';
+import '@vaadin/vaadin-lumo-styles/style.js';
+import '@vaadin/vaadin-lumo-styles/typography.js';
+import '@vaadin/vaadin-lumo-styles/mixins/required-field.js';
+import '@vaadin/vaadin-lumo-styles/font-icons.js';
+import '@vaadin/vaadin-lumo-styles/mixins/field-button.js';
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+
+const $_documentContainer = html`<dom-module id="lumo-text-field" theme-for="my-field">
```

Comment:
Not sure if this is intentional, but it will conflict with the official Vaadin Text Field component style sheet, if both are imported to the same app. Prefer using unique IDs instead.

```suggestion
const $_documentContainer = html`<dom-module id="my-text-field" theme-for="my-field">
```

Reply:
Note, that if we change to using `registerStyles` (see other thread), this is no longer relevant (no ID necessary).

Reply:
This component is just here for testing purposes as we don't want Flow to depend on the components so its basically a ripped copy of the textField.

Reply:
I would still change it to use `registerStyles`, to future-proof it if this test is ever converted to using LitElement instead of PolymerElement.

Reply:
Do we already have a component implementation using registerStyles internally?
As this should test that the feature works with the components.

Reply:
@caalador https://github.com/vaadin/vaadin-app-layout uses registerStyles for example in the branch `next`

Reply:
The big question is just that should we use here in the test something that we do not use in the actual version where the feature is coming out?

Reply:
I'm the wrong person to decide that ;) Just wanted to make sure, that theming is tested with both worlds because lit element is planned for V14.5 and V18.

Reply:
registerStyles is an officially supported part of the API already in 14.x series. All components support it.

Reply:
The implementation of our components for 14.x is still Polymer 2 based (because of the need to support Bower-mode), and the automatic conversion to Polymer 3 generates the uglier version of the style modules, which is used here as well (`$_documentContainer = ...`).